### ISSUE TITLE
Status of the project?

### ISSUE THREAD

**Issue by cniska** [elierotenberg/react-prepare#3]
@elierotenberg I was actually going to start writing a similar library myself for my own needs when I stumbled across your library. What is the status of this project?

**Comment by elierotenberg** [comment]
Hello @crisu83, and thanks for your interest :)
Atm the project is getting a bit dusty, but I plan on cleaning it up and promoting it, as I firmly believe this approach is the best (ie. most idiomatic and efficient) approach to do isomorphic data-fetching in a React app.

Are there missing features you would like to have in react-prepare?

**Comment by cniska** [comment]
I'm currently adding server-side-rendering to our [company boilerplate](https://github.com/nordsoftware/react-boilerplate) and right now I'm using a library I wrote myself a while back called [redux-fetch-data](https://www.npmjs.com/package/redux-fetch-data). 

The motivation behind that library is the same, but I'm starting to see the big picture here, which wasn't clear back then. That's why I was thinking of writing a new library, but now I see you already wrote one that might suite my needs.

I'll have to give the library a try and a pair of second eyes before I say more. It would help if you'd look at my old implementation and explain how I can transform that approach to use this library. 

In other words it was not clear to me after reading the readme how to actually use the library to prepare some content for a component. A simple example where one resolves a promise before rendering on the server would suffice.

**Comment by cniska** [comment]
I would like to suggest adding Flowtype type definitions to the project and use `const` to define functions for a cleaner syntax:

```
const prepared = (prepare, {...}) => (WrappedComponent) => {
  ...
}
```
vs
```
function prepared(prepare, {...}) {
  return (OriginalComponent) => {
    ...
  }
}
```

**Comment by cniska** [comment]
@elierotenberg thoughts?

**Comment by elierotenberg** [comment]
Hello @crisu83,

As promised, I just did a pass of dust-cleaning on the project (upgraded deps and stuff).

For a real-world-like example of using the library, see https://github.com/elierotenberg/react-prepare/blob/master/src/tests/dispatched.spec.js.
In this full-fledged example, you can see examples of components which dispatch an action (to fetch data) before rendering.

I'll consider adding Flow annotations, but I'd be pleased to read your thoughts in the meantime.
